common/osqp_interface/src/osqp_interface.cpp

```diff
@@ -68,6 +68,11 @@ OSQPInterface::OSQPInterface(
   initializeProblem(P, A, q, l, u);
 }
 
+OSQPInterface::~OSQPInterface(){
+  free(m_data->P);
```

Comment:
Are we sure m_data->P or even m_data has been allocated by the time the destructor is called?
Calling free on unallocated data is UB, so I guess it is safer to add a `if(m_data && m_data->P)`?

Reply:
You are right. One constructor (in addition to the default constructor) do not call `initializeProblem` which could lead to UB when calling the destructor.
I updated the code to fix that issue.